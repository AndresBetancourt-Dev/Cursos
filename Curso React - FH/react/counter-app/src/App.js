import React, { Fragment } from "react";

//Functional Compontent

const App = (props) => {
  console.log(props);

  const title = `High Availabilty`;
  const message = `The use of the best technologies, Node.js, React & Redux`;
  const utils = {
    sideText: "Wil be forever",
    elementBackground: "image",
    enabled: false,
    React: true,
    Java: true,
    Redux: false,
    SpringBoot: props.info,
  };

  return (
    <Fragment>
      <h1>{title}</h1>
      <h4>{message}</h4>
      <pre>{JSON.stringify(utils, null, 4)}</pre>
    </Fragment>
  );
};

export default App;
