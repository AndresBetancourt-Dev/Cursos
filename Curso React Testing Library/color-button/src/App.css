body{
  background-color: #282c34;
}

.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

button{
  padding: 1rem 2rem;
  outline: none;
  border: none;
  font-weight: bold;
  position: relative;
  color: white;
  isolation: isolate;
}

button::before{
  content: '';
  position: absolute;
  width: 100%;
  height: 100%;
  background: radial-gradient(rgba(0,0,0,0.25), transparent);
  z-index: -1;
  top: 0;
  left: 0;
}