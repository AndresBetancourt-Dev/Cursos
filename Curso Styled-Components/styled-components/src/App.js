import "styled-components/macro";
import styled, { css } from "styled-components/macro";
import React from "react";
import { GlobalStyles } from "./styles/global.styles";
import Counter from "./components/Counter.new";

const Button = styled.button`
  color: palevioletred;
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
  cursor : pointer;
`;

// A new component based on Button, but with some override styles
const TomatoButton = styled(Button)`
  color: tomato;
  border-color: tomato;
`;

// Create a Title component that'll render an <h1> tag with some styles
const Title = styled.h1`
  font-size: 1.5em;
  text-align: center;
  color: palevioletred;
`;

// Create a Wrapper component that'll render a <section> tag with some styles
const Wrapper = styled.section`
  padding: 4em;
  background: papayawhip;
`;


const ReversedButton = props => <Button {...props} children={props.children.split('').reverse()} />

// This could be react-router-dom's Link for example
const Link = ({ className, children }) => (
  <a className={className}>
    {children}
  </a>
);

const StyledLink = styled(Link)`
  color: palevioletred;
  font-weight: bold;
`;

const Input = styled.input`
  padding: 0.5em;
  margin: 0.5em;
  color: ${props => props.inputColor || "palevioletred"};
  background: papayawhip;
  border: none;
  border-radius: 3px;
`;

const App = () => {
  return (
    <div style={{ padding: "2rem" }}>
      <GlobalStyles />
      <Wrapper>
        <Title>Hello World!</Title>
      </Wrapper>
      <Button>Normal</Button>
      <TomatoButton>Primary</TomatoButton>
      <Button as="a" href="#">Link with Button styles</Button>
    <TomatoButton as="a" href="#">Link with Tomato Button styles</TomatoButton>
    <Button as={ReversedButton}>Custom Button with Normal Button styles</Button>
    <Link>Unstyled, boring Link</Link>
    <br />
    <StyledLink>Styled, exciting Link</StyledLink>
    <Input defaultValue="@probablyup" type="text" />
    <Input defaultValue="@geelen" type="text" inputColor="rebeccapurple" />
    <Counter/>
    </div>
  );
};

export default App;
