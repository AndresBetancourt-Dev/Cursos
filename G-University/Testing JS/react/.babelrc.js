module.exports = {
  presets: [
    ['@babel/preset-env', {targets: {node: 'current'}}],
    [
      "@babel/preset-react",
      {
        development: process.env.BABEL_ENV === "development",
      },
    ]
  ],
};
