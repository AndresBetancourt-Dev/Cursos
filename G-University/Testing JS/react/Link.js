import React from "react";

const STATUS = {
  HOVERED: "hovered",
  NORMAL: "normal",
};

const Link = ({ page, children }) => {
  const [status, setStatus] = React.useState(STATUS.NORMAL);

  const onMouseEnter = () => {
    setStatus(STATUS.HOVERED);
  };

  const onMouseLeave = () => {
    setStatus(STATUS.NORMAL);
  };

  return (
    <a
      className={status}
      href={page || "#"}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
      {children}
    </a>
  );
};

export default Link;
