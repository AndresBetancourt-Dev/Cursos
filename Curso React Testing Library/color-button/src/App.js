import { useState } from 'react';
import './App.css';

export const CHANGE_TO = 'Change to ';
export const RED = 'red';
export const BLUE = 'blue';
export const GREY = 'grey';

export const replaceCamelCaseWithSpaces = (colorName) => {
  return colorName.replace(/\B([A-Z])\B/g,' $1');
}

function App() {

  const [buttonColor, setButtonColor] = useState(RED);
  const [disabled,setDisabled] = useState(false);
  const newButtonColor = buttonColor === RED ? BLUE : RED;
 
  return (
    <div className="App">
      <button style={{backgroundColor: disabled ? GREY : buttonColor}} onClick={()=> setButtonColor(newButtonColor)} disabled={disabled}>{CHANGE_TO}{newButtonColor}</button>
      <input type={"checkbox"} defaultChecked={disabled} aria-labelledby="enable-button-checkbox" aria-checked={disabled} onClick={(e)=> setDisabled(e.target.checked)}/>
      <label id="enable-button-checkbox">Disable button</label>
    </div>
  );
}

export default App;
