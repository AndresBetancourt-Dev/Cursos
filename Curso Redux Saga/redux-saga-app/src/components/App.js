import { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { Alert } from "reactstrap";
import {
  createUserRequest,
  deleteUserRequest,
  getUsersRequest,
  usersError,
} from "../actions/users";
import NewUserForm from "./NewUserForm";
import UsersList from "./UsersList";

function App() {
  const users = useSelector((state) => state.users.items);
  const error = useSelector((state) => state.users.error);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getUsersRequest());
  }, []);

  const handleSubmit = ({ firstName, lastName }) => {
    dispatch(createUserRequest({ firstName, lastName }));
  };

  const handleDelete = (id) => {
    dispatch(deleteUserRequest(id));
  };

  const handleCloseAlert = () => {
    dispatch(usersError({ error: "" }));
  };

  return (
    <div style={{ margin: "0 auto", padding: "20px", maxWidth: "600px" }}>
      <Alert color="danger" isOpen={Boolean(error)} toggle={handleCloseAlert}>
        {error}
      </Alert>
      <NewUserForm onSubmit={handleSubmit} />
      <UsersList users={users} onDeleteUser={handleDelete} />
    </div>
  );
}

export default App;
