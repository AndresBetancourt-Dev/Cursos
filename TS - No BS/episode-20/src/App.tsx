import { useCallback, useEffect, useState } from 'react';
import './App.css';
import { Box } from './components/Box/Box';
import { Heading } from './components/Heading';
import { Incrementer } from './components/Incrementer';
import { List } from './components/List';
import Todo from './components/Todo';
interface Payload {
  text: string
}

function App() {
  const [payload, setPayload] = useState<Payload | null>(null);
  const [value, setValue] = useState<number>(0)

  useEffect(() => {
    fetch('/data.json').then((response) => response.json()).then(data => setPayload(data))
  }, [])

  const onListClick = useCallback((item: string) => {
    alert(item)
  }, []);

  return (
    <div >
      <Heading title="Introduction" center />
      <Box>
        Pepito perez
      </Box>
      <List items={['Julio', 'Sanchez', 'Cristo']} onClick={onListClick} />
      <Box>
        {JSON.stringify(payload)}
      </Box>
      <Todo />
      <Incrementer value={value} setValue={setValue}
      />
    </div>
  );
}

export default App;
