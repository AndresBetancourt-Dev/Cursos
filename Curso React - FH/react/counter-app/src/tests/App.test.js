import { describe, test, expect } from "@jest/globals";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import "@testing-library/jest-dom/extend-expect";
import React from "react";
import App from "../App";
import { shallow } from "enzyme";

describe("React App Component tests", () => {
  test("Title", () => {
    let text = "React the best";
    const { getByText } = render(
      <App
        info="Need to learn Springboot, ASAP"
        react={{ title: text }}
        java={true}
        redux={{ title: { reduxTitle: "" } }}
      />
    );
    expect(getByText(text)).toBeInTheDocument();
  });

  test("title prop", () => {
    const wrapper = shallow(
      <App
        info="Need to learn Springboot, ASAP"
        react={{ title: "React the best" }}
        java={true}
        redux={{ title: { reduxTitle: "" } }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar el subtitulo enviado por props", () => {
    const subtitle = "This is my new great subtitle.";
    const wrapper = shallow(
      <App
        info="Need to learn Springboot, ASAP"
        react={{}}
        java={true}
        redux={{ title: { reduxTitle: "" } }}
        subtitle={subtitle}
      />
    );

    const subtitleRendered = wrapper.find("h2").text();

    expect(subtitleRendered).toStrictEqual(subtitle);
  });
});
