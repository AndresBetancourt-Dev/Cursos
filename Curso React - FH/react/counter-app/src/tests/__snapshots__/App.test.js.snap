// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React App Component tests title prop 1`] = `
<Fragment>
  <h1>
    React the best
  </h1>
  <h2>
    I will be the best programmer
  </h2>
  <h3>
    Need to learn Springboot, ASAP
  </h3>
  <h4>
    The use of the best technologies, Node.js, React & Redux
  </h4>
  <pre>
    {
    "sideText": "Wil be forever",
    "elementBackground": "image",
    "enabled": false,
    "React": "React the best",
    "Java": "Will course the spring boot course",
    "Redux": "",
    "SpringBoot": "Need to learn Springboot, ASAP"
}
  </pre>
</Fragment>
`;
